### ISSUE TITLE
Cargo Build failing due to typo: Config has no field named dedicated_treshold

### ISSUE THREAD

**Issue by sotrh** [gfx-rs/wgpu#1288]
**Description**
I created a new project and added version 0.7 of wgpu to it. I went to build it and got the following error when it tried to build wgpu-core.

```
error[E0560]: struct `Config` has no field named `dedicated_treshold`
  --> /home/benjamin/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-core-0.7.1/src/device/alloc.rs:18:13
   |
18 |             dedicated_treshold: 32 << 20,
   |             ^^^^^^^^^^^^^^^^^^ help: a field with a similar name exists: `dedicated_threshold`

error[E0560]: struct `Config` has no field named `preferred_dedicated_treshold`
  --> /home/benjamin/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-core-0.7.1/src/device/alloc.rs:19:13
   |
19 |             preferred_dedicated_treshold: 8 << 20,
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a field with a similar name exists: `preferred_dedicated_threshold`

error[E0560]: struct `Config` has no field named `transient_dedicated_treshold`
  --> /home/benjamin/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-core-0.7.1/src/device/alloc.rs:20:13
   |
20 |             transient_dedicated_treshold: 128 << 20,
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a field with a similar name exists: `transient_dedicated_threshold`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0560`.
error: could not compile `wgpu-core`

To learn more, run the command again with --verbose.
```

**Repro steps**
Create a new crate and add `wgpu = "0.7.0"` to your Cargo.toml. Then run `cargo build`. You don't have to write any wgpu code.

**Expected vs observed behavior**
You should be able to build the version of wgpu deployed to crates.io but currently the build fails without the user writing any wgpu code.

**Extra materials**

If you look at the [alloc.rs file](https://github.com/gfx-rs/wgpu/blob/master/wgpu-core/src/device/alloc.rs) you can see that the typo is on the master branch.

**Platform**
wgpu version 0.7
Linux Mint (Ubuntu/Debian)
rustc 1.51.0 (2fd73fabe 2021-03-23)
cargo 1.51.0 (43b129a20 2021-03-16)

**EDIT**
Cloning the wgpu repo and running `cargo build` works. It seems cargo is favoring v0.3.0 instead of the most recent version v0.3.1 when building from source.

```
benjamin@sotrh-top:~/dev/wgpu$ cargo build
  Downloaded gpu-alloc v0.3.0
  Downloaded 1 crate (15.9 KB) in 0.42s
    Finished dev [unoptimized + debuginfo] target(s) in 0.81s
benjamin@sotrh-top:~/dev/wgpu$ 
```
This could be a bug with cargo

**Comment by Gordon-F** [comment]
Fixed by zakarumych/gpu-alloc#39. Remove your current `Cargo.lock` and try to build again. Should fix all issues.

**Comment by sotrh** [comment]
Ok that's weird. It was pulling v0.3.1 of gpu alloc for some reason. It was yanked about 4hrs ago so maybe that was the issue. It's building now though.
